codalab/rest/groups.py

```diff
@@ -0,0 +1,167 @@
+"""
+Worksheets REST API Groups Views.
+"""
+import httplib
+
+from bottle import abort, get, delete, post, request, local
+from marshmallow_jsonapi import Schema, fields
+
+from codalab.lib.server_util import (
+    json_api_include,
+    query_get_list,
+)
+from codalab.lib.spec_util import validate_uuid, validate_name
+from codalab.rest.users import UserSchema
+from codalab.rest.util import get_group_info, ensure_unused_group_name
+from codalab.server.authenticated_plugin import AuthenticatedPlugin
+from codalab.server.json_api_plugin import JsonApiPlugin
+
+#############################################################
+#  GROUP DE/SERIALIZATION AND VALIDATION SCHEMAS
+#############################################################
+
+
+class GroupSchema(Schema):
+    id = fields.String(validate=validate_uuid, attribute='uuid')
+    name = fields.String(required=True, validate=validate_name)
+    user_defined = fields.Bool(dump_only=True)
+    owner = fields.Relationship(include_data=True, type_='users', attribute='owner_id')
+    memberships = fields.Relationship(dump_only=True, many=True, type_='group-memberships', include_data=True)
+
+    class Meta:
+        type_ = 'groups'
+
+
+class GroupMembershipSchema(Schema):
+    id = fields.Integer(as_string=True)
+    user = fields.Relationship(required=True, include_data=True, type_='users', attribute='user_id')
+    group = fields.Relationship(required=True, include_data=True, type_='groups', attribute='group_uuid')
+    is_admin = fields.Boolean(required=True)
+
+    class Meta:
+        type_ = 'group-memberships'
+
+
+#############################################################
+#  GROUP REST API ENDPOINTS
+#############################################################
+
+
+@get('/groups/<group_spec>', apply=[AuthenticatedPlugin(), JsonApiPlugin()])
+def fetch_group(group_spec):
+    """Fetch a single group."""
+    group = get_group_info(group_spec, need_admin=False)
+    document = GroupSchema().dump(group).data
+    include_group_relationships(document, [group])
+    return document
+
+
+@get('/groups', apply=[AuthenticatedPlugin(), JsonApiPlugin()])
+def fetch_groups():
```

Comment:
Wow, I have no idea why you're putting in all this time to make this code so generic and flexible. Why support have include-memberships and users be False?